### PR TITLE
Misc line length, indentation adjustments

### PR DESCRIPTION
**Pull request by tkelman**
Should be no functional change here, been accumulating these on a branch again. View as https://github.com/JuliaLang/julia/pull/21049/files?w=1 for smaller diff

### CODE REVIEW

File: base/abstractarray.jl [hunk 1]
```diff
@@ -1152,7 +1182,8 @@ end
     _cshp(tail(dims), (out..., next), tail(shape), tail(nshape))
 end
 
-_cs(d, concat, a, b) = concat ? (a + b) : (a == b ? a : throw(DimensionMismatch(string("mismatch in dimension ", d, " (expected ", a, " got ", b, ")"))))
+_cs(d, concat, a, b) = concat ? (a + b) : (a == b ? a : throw(DimensionMismatch(string(
+    "mismatch in dimension ", d, " (expected ", a, " got ", b, ")"))))
```
Comment (ararslan) [inline base/abstractarray.jl]: I'd probably move the entire `throw` to this line. Looks a little funny otherwise.

File: base/linalg/lu.jl [hunk 2]
```diff
@@ -78,7 +78,9 @@ function generic_lufact!{T,Pivot}(A::StridedMatrix{T}, ::Type{Val{Pivot}} = Val{
 end
 
 # floating point types doesn't have to be promoted for LU, but should default to pivoting
-lufact{T<:AbstractFloat}(A::Union{AbstractMatrix{T},AbstractMatrix{Complex{T}}}, pivot::Union{Type{Val{false}}, Type{Val{true}}} = Val{true}) = lufact!(copy(A), pivot)
+lufact{T<:AbstractFloat}(A::Union{AbstractMatrix{T},AbstractMatrix{Complex{T}}},
```
Comment (ararslan) [inline base/linalg/lu.jl]: A space after the first `AbstractMatrix{T},` would be nice (though it wasn't already there so nbd)

File: test/parse.jl [hunk 3]
```diff
@@ -246,7 +246,8 @@ for T in vcat(subtypes(Signed), subtypes(Unsigned))
 
     # Test that the entire input string appears in error messages
     let s = "     false    true     "
-        result = @test_throws ArgumentError get(Base.tryparse_internal(Bool, s, start(s), endof(s), 0, true))
+        result = @test_throws(ArgumentError,
```
Comment (ararslan) [inline test/parse.jl]: I think I'd leave this one as it was. It's no longer than base/sysimg.jl line 145, for example, and is a little easier to follow when it's on one line (IMO).

File: test/perf/kernel/go_benchmark.jl [hunk 4]
```diff
@@ -91,261 +91,261 @@ POS(board::Board, i::Int, j::Int) = (j - 1) * board.size + i
...
+    end
 
-  # Illegal ko recapture. It is not illegal to fill the ko so we must
-  # check the color of at least one neighbor.
-  if i == board.ko_i && j == board.ko_j && ((on_board(board, i - 1, j) && board[i - 1, j] == other) || (on_board(board, i + 1, j) && board[i + 1, j] == other))
-    return false
-  end
+    # Illegal ko recapture. It is not illegal to fill the ko so we must
+    # check the color of at least one neighbor.
+    if i == board.ko_i && j == board.ko_j && ((on_board(board, i - 1, j) && board[i - 1, j] == other) || (on_board(board, i + 1, j) && board[i + 1, j] == other))
```
Comment (ararslan) [inline test/perf/kernel/go_benchmark.jl]: Really long line here, might be nice to break it up. Maybe a line break after the last `||`?

File: base/abstractarray.jl (hunk 1 above)
Comment (tkelman) [inline base/abstractarray.jl]: the line would be 102 chars wide if we did that

File: test/parse.jl (hunk 3 above)
Comment (tkelman) [inline test/parse.jl]: 112 is too wide - I'd break up the StridedArray aliases further if there were decent places to do so

File: base/abstractarray.jl (hunk 1 above)
Comment (ararslan) [inline base/abstractarray.jl]: Oh, sad.

File: test/parse.jl (hunk 3 above)
Comment (ararslan) [inline test/parse.jl]: ok

**Review by ararslan — APPROVED** [review]

**Pull request merged**